tests/e2e/test_async_client.py

```diff
@@ -175,14 +175,14 @@ async def test_invalid_job_id_error(self):
             "Can't build ExecutionStatusResponse from "
             "{'error': 'The requested execution ID (ID: Wonky Job ID) is invalid.'}",
         )
-        await dune.close_session()
+        await dune.disconnect()
 
     async def test_disconnect(self):
         dune = AsyncDuneClient(self.valid_api_key)
         await dune.connect()
         results = (await dune.refresh(self.query)).get_rows()
         self.assertGreater(len(results), 0)
-        await dune.close_session()
+        await dune.disconnect()
         self.assertTrue(cl._session.closed)
```

Comment:
there should be `self.assertTrue(dune._session.closed)`, my bad

but getting `Too many requests in client`  is interesting, not sure whats the best way to approach it

Reply:
Maybe it's retrying too much I'll try it out tomorrow.